CMakeLists.txt

```diff
@@ -99,6 +99,13 @@ add_subdirectory(drivers/source/usb EXCLUDE_FROM_ALL)
 add_subdirectory(features EXCLUDE_FROM_ALL)
 add_subdirectory(platform/FEATURE_EXPERIMENTAL_API EXCLUDE_FROM_ALL)
 
+if(${MBED_BAREMETAL_PROFILE})
```

Comment:
I think it is more likely that a board does not support the full profile than it does not support the baremetal profile.
In any case, you should check that the rtos profile is supported if `MBED_BAREMETAL_PROFILE` is false and also that the baremetal profile is supported if `MBED_BAREMETAL_PROFILE` is true.

Also, please add a comment above this, something like:
`Verify that the selected application OS profile is supported by the Mbed target`